src/data-sources/mongodb/mongodb-data-source.js

```diff
@@ -113,32 +146,90 @@ class MongoDBDataSource extends DataSource {
     }
   }
 
+  /**
+   * Transform a single filter into its MongoDB counterpart.
+   *
+   * @param {object} filter - The filter object to transform.
+   * @returns {object} The MongoDB filter object.
+   */
+  #transformSingleFilter(filter) {
+    if (!filter.key) return {};
+
+    const valueToArray = (value) => {
+      if (typeof value === 'string') {
+        return value.split(',');
+      } else if (Array.isArray(value)) {
+        return value;
+      } else {
+        return [];
+      }
+    };
+
+    switch (filter.operator) {
+      case '==':
+        return { [filter.key]: filter.value };
+      case '!==':
+        return { [filter.key]: { $ne: filter.value } };
+      case 'in':
+        return { [filter.key]: { $in: valueToArray(filter.value) } };
+      case 'not-in':
+        return { [filter.key]: { $nin: valueToArray(filter.value) } };
+      default:
+        return { [filter.key]: filter.value };
+    }
+  }
+
+  /**
+   * Transform an array of filters into a MongoDB filter query.
+   *
+   * @param {Array<object>} filters - An array of filter objects to transform.
+   * @returns {object} The MongoDB filter query.
+   * @throws Will throw an error if a single filter transformation fails.
+   */
   #transformFilters(filters) {
     try {
-      const transformedFilters = {};
+      const transformedFilters = { $and: [] };
 
-      if (filters && filters.length > 1) {
-        transformedFilters['$and'] = [];
+      filters.forEach((filter) => {
+        if (!filter.value) return;
 
-        for (const filter of filters) {
-          if (filter.key) {
-            transformedFilters['$and'].push({
-              [filter.key]: filter.value,
-            });
-          }
+        if (filter.key !== 'skip' && filter.key !== 'limit') {
+          transformedFilters.$and.push(this.#transformSingleFilter(filter));
         }
-      } else if (filters && filters.length === 1) {
-        const filter = filters[0];
-        transformedFilters[filter.key] = filter.value;
-      }
+      });
 
-      return transformedFilters;
+      return transformedFilters.$and.length ? transformedFilters : {};
     } catch (error) {
-      this._console.error(error);
-
+      this._console.error('Error transforming filter: ', error);
       return {};
     }
   }
+
+  /**
+   * Extract pagination data from the given filter array.
+   *
+   * @param {Array<object>} filters - An array of filter objects.
+   * @returns {object} An object containing the 'skip' and 'limit' pagination values.
+   * @throws Will throw an error if pagination values are not valid numbers.
+   */
+  #extractPagination(filters) {
+    const pagination = {
+      skip: null,
```

Comment:
Null is required, as in case the URL does not include any 'skip' or 'limit' fields, a validation will be triggered, and it will return all the data in the collection.